src/java_bytecode/expr2java.cpp

```diff
@@ -510,13 +517,8 @@ std::string expr2javat::convert(
   }
   else if(src.id()==ID_java_string_literal)
     return '"'+MetaString(src.get_string(ID_value))+'"';
-  else if(src.id()==ID_constant && (type.id()==ID_bool || type.id()==ID_c_bool))
-  {
-    if(src.is_true())
-      return "true";
-    else
-      return "false";
-  }
+  else if(src.id()==ID_constant)
+    return convert_constant(to_constant_expr(src), precedence);
```

Comment:
It's a somewhat wild guess: is this part of the change causing the regression test to fail (the regression test now has an extra set of `(`...`)` around `"java::D"`). Also I think that the `ID_nullptr` case should be moved to `convert_constant`.